tests/ExecutorTest.php

```diff
@@ -329,22 +329,61 @@ public function provideScenarios(): array
                 'startCommand' => 'cp /tmp/code.tar.gz /mnt/code/code.tar.gz && nohup helpers/start.sh "pm2 start src/server.js --no-daemon"',
                 'buildCommand' => 'tar -zxf /tmp/code.tar.gz -C /mnt/code && helpers/build.sh "npm i"',
                 'assertions' => function ($response) {
-                    $this->assertEquals(500, $response['headers']['status-code']);
-                    $this->assertStringContainsString('Invalid response. This usually means too large logs or errors', $response['body']['message']);
-                }
+                    $this->assertEquals(200, $response['headers']['status-code']);
+                    $this->assertEquals("OK", $response['body']['body']);
+                    $this->assertLessThanOrEqual(5 * 1024 * 1024, strlen($response['body']['logs']));
+                    $this->assertLessThanOrEqual(5 * 1024 * 1024, strlen($response['body']['errors']));
+                },
+                'body' => function () {
+                    return 1;
+                },
             ],
             [
-                'image' => 'openruntimes/node:v3-18.0',
+                'image' => 'openruntimes/node:v4-18.0',
+                'entrypoint' => 'index.js',
+                'folder' => 'node-logs',
+                'version' => 'v4',
+                'startCommand' => 'cp /tmp/code.tar.gz /mnt/code/code.tar.gz && nohup helpers/start.sh "pm2 start src/server.js --no-daemon"',
+                'buildCommand' => 'tar -zxf /tmp/code.tar.gz -C /mnt/code && helpers/build.sh "npm i"',
+                'assertions' => function ($response) {
+                    $this->assertEquals(200, $response['headers']['status-code']);
+                    $this->assertEquals("OK", $response['body']['body']);
+                    $this->assertLessThanOrEqual(5 * 1024 * 1024, strlen($response['body']['logs']));
+                    $this->assertLessThanOrEqual(5 * 1024 * 1024, strlen($response['body']['errors']));
+                },
+                'body' => function () {
+                    return 5;
+                },
+            ],
+            [
+                'image' => 'openruntimes/node:v4-18.0',
+                'entrypoint' => 'index.js',
+                'folder' => 'node-logs',
+                'version' => 'v4',
+                'startCommand' => 'cp /tmp/code.tar.gz /mnt/code/code.tar.gz && nohup helpers/start.sh "pm2 start src/server.js --no-daemon"',
+                'buildCommand' => 'tar -zxf /tmp/code.tar.gz -C /mnt/code && helpers/build.sh "npm i"',
+                'assertions' => function ($response) {
+                    $this->assertEquals(200, $response['headers']['status-code']);
+                    $this->assertEquals("OK", $response['body']['body']);
+                    $this->assertGreaterThan(1 * 1024 * 1024, strlen($response['body']['logs']));
+                    $this->assertLessThanOrEqual(1 * 1024 * 1024, strlen($response['body']['errors']));
```

Comment:
Here, I would test it's larger than 5MB, but smaller than 6MB. I would also test that our warning at the end is there.